src/medley/core.cljc

```diff
@@ -80,54 +90,66 @@
   "Maps a function over the key/value pairs of an associate collection. Expects
   a function that takes two arguments, the key and value, and returns the new
   key and value as a collection of two elements."
-  [f coll]
-  (reduce-map (fn [xf] (fn [m k v] (let [[k v] (f k v)] (xf m k v)))) coll))
+  ([f]
+   (completing-rf (fn [rf result [k v]] (rf result (f k v)))))
```

Comment:
Isn't this just `(map f)`?

Reply:
It's `(map (fn [[k v]] (f k v)))`, or more loosely `(map #(apply f %))`